hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/HoodieBaseRelation.scala

```diff
@@ -470,10 +481,8 @@ abstract class HoodieBaseRelation(val sqlContext: SQLContext,
       usesVirtualKeys = !tableConfig.populateMetaFields(),
       recordPayloadClassName = tableConfig.getPayloadClass,
       metadataConfig = fileIndex.metadataConfig,
-      mergerImpls = optParams.getOrElse(HoodieWriteConfig.MERGER_IMPLS.key(),
-        HoodieWriteConfig.MERGER_IMPLS.defaultValue()),
-      mergerStrategy = optParams.getOrElse(HoodieWriteConfig.MERGER_STRATEGY.key(),
-        metaClient.getTableConfig.getMergerStrategy)
+      mergerImpls = mergerImpls,
```

Comment:
Leaving comments here for `generateRecordMerger` as well:

 - Let's do config parsing here, passing in `Seq[String]` instead of `String` 
 - Let's rename `generateRecordMerger` to `createRecordMerger`